src/OrchardCore.Modules/OrchardCore.Shortcodes/Controllers/AdminController.cs

```diff
@@ -34,6 +35,8 @@ public class AdminController : Controller
         private readonly IHtmlLocalizer H;
         private readonly dynamic New;
 
+        private const string NameValidationRegex = "[a-zA-Z$_][a-zA-Z0-9$_]*";
```

Comment:
Can we do this without regex? 

Preference is char matching (both performance, and the issue of unconstrained regex)

I would have to look at shortcodes, but probably you can just lean on the Shortcodes/Parlot `ReadIdentifier` and compare the sequence.

Reply:
I've realised my regex needed start and end anchors, so it should have been "^[a-zA-Z$_][a-zA-Z0-9$_]*$".

I've had a go at using parlot, please see my latest push to see if it's okay?

Reply:
Yeah looks much better to me. can you remove the regex?

Reply:
Doh! Yes, I should not have left that there. Taken it out.